COMOPT-417, Product Container Documentations (#325)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/recommendations/containers/product-list.mdx

---

import Diagram from '@components/Diagram.astro';
import OptionsTable from '@components/OptionsTable.astro';

> The `ProductList` container manages and displays a list of recommended products based on the current product. Its behavior is driven by configuration options such as `currentSku`, pageType and routing via routeProduct callbacks.

Reviewer: ```suggestion
The `ProductList` container manages and displays a list of recommended products based on the current product. Its behavior is driven by configuration options such as `currentSku`, `pageType`, and `routing` through `routeProduct` callbacks.
```

    ['Option', 'Type', 'Req?', 'Description'],
    ['currentSku', 'string', 'Yes', 'Current SKU for recommendation context.'],
    ['cartSkus', 'string', 'array', 'Cart SKUs for recommendation context.'],
    ['userPurchaseHistory', 'array', 'No', 'User Purchase History for recommendation context.'],
    ['userViewHistory', 'array', 'No', 'User View History for recommendation context.'],
    ['routeProduct', 'function', 'No', 'Callback function that returns a product URL.'],
    ['pageType', 'function', 'No', 'Page type where recommendations are rendered (e.g., Product, Cateogy, Cart)'],

>     ['hideHeading', 'function', 'No', 'Hide recommendations heading text'],

Reviewer: ```suggestion
    ['hideHeading', 'function', 'No', 'Hide recommendations heading text.'],
```

>     ['slots', 'function', 'No', 'Slots for default heading, thumbnail, price and footer'],

Reviewer: ```suggestion
    ['slots', 'function', 'No', 'Slots for default heading, thumbnail, price, and footer.'],
```

  ]}
/>

## Example configuration

> The following example demonstrates how to render the `ProductList` container with the `currentSku`,`pageType` and `routeProduct` callbacks with different slots like default heading, thumbnail, price and footer:

Reviewer: ```suggestion
The following example demonstrates how to render the `ProductList` container with the `currentSku`, `pageType`, and `routeProduct` callbacks with different slots like default heading, thumbnail, price, and footer:
```

Reviewer: According to the table above, `currentSku` is a string; not a callback (function).
